sklearn/datasets/openml.py

```diff
@@ -0,0 +1,240 @@
+import json
+import numbers
+import sys
+import os
+import pandas as pd
```

Comment:
No, we can't (yet) have this kind of dependency on pandas. We have an optional dependency in testing on pandas, and we could return DataFrames in the case that the user asks for one explicitly with a parameter. Otherwise, in the world scikit-learn currently lives in, we need to be dealing in homogeneous arrays and sparse matrices.

Reply:
We could potentially consider returning DataFrames by default, but it would have to be a local import.